txnsync/outgoing_test.go

```diff
@@ -0,0 +1,381 @@
+				transactions.SignedTxn{
+					Txn: transactions.Transaction{
+						Type: protocol.UnknownTx,
+					},
+				},
+			},
+		},
+	}
+
+	asyncEncoder := messageAsyncEncoder{
+		state: &s,
+		messageData: sentMessageMetadata{
+			message: &transactionBlockMessage{
+				Version: txnBlockMessageVersion,
+				Round:   1,
+			},
+			transactionGroups: txnGrps,
+			peer:              &Peer{},
+		},
+		roundClock: timers.MakeMonotonicClock(time.Now()),
+	}
+
+	asyncEncoder.state.messageSendWaitGroup.Add(1)
+
+	err := asyncEncoder.asyncEncodeAndSend(nil)
+
+	a.Nil(err)
+	a.True(warnCalled)
+	a.True(sendPeerMessageCalled)
+
+}
+
+// TestAsyncEncodeAndSend Tests that SendPeerMessage is called in the node connector
+func TestAsyncEncodeAndSend(t *testing.T) {
+	a := require.New(t)
+
+	var s syncState
+	s.clock = timers.MakeMonotonicClock(time.Now())
+	s.log = mockAsyncLogger{}
+	sendPeerMessageCalled := false
+	s.node = mockAsyncNodeConnector{called: &sendPeerMessageCalled}
+	s.messageSendWaitGroup = sync.WaitGroup{}
+
+	asyncEncoder := messageAsyncEncoder{
+		state: &s,
+		messageData: sentMessageMetadata{
+			message: &transactionBlockMessage{
+				Version: txnBlockMessageVersion,
+				Round:   1,
+			},
+			peer: &Peer{},
+		},
+		roundClock: timers.MakeMonotonicClock(time.Now()),
+	}
+
+	asyncEncoder.state.messageSendWaitGroup.Add(1)
+
+	err := asyncEncoder.asyncEncodeAndSend(nil)
+
+	a.Nil(err)
+	a.True(sendPeerMessageCalled)
+
+}
+
+// TestAssemblePeerMessage1 Tests assemblePeerMessage with messageConstBloomFilter msgOps
+func TestAssemblePeerMessage1(t *testing.T) {
+	a := require.New(t)
+
+	s := syncState{clock: timers.MakeMonotonicClock(time.Now())}
+
+	s.profiler = makeProfiler(1*time.Millisecond, s.clock, s.log, 1*time.Millisecond)
+
+	peer := Peer{}
+
+	pendingTransactions := pendingTransactionGroupsSnapshot{
+		pendingTransactionsGroups: []transactions.SignedTxGroup{
+			transactions.SignedTxGroup{},
+		},
+	}
+
+	peer.setLocalRequestParams(111, 222)
+	peer.lastReceivedMessageTimestamp = 100
+	peer.lastReceivedMessageLocalRound = s.round
+
+	expectedFilter := s.makeBloomFilter(requestParams{Offset: 111, Modulator: 222}, pendingTransactions.pendingTransactionsGroups, &s.lastBloomFilter)
+
+	s.isRelay = true
+	peer.isOutgoing = true
+	peer.state = peerStateLateBloom
+
+	metaMessage := s.assemblePeerMessage(&peer, &pendingTransactions)
+
+	a.Equal(metaMessage.message.UpdatedRequestParams.Modulator, byte(222))
+	a.Equal(metaMessage.message.UpdatedRequestParams.Offset, byte(111))
+	a.Equal(metaMessage.peer, &peer)
+	a.Equal(metaMessage.message.Version, int32(txnBlockMessageVersion))
+	a.Equal(metaMessage.message.Round, s.round)
+	a.True(metaMessage.message.MsgSync.ResponseElapsedTime != 0)
+	a.Equal(s.lastBloomFilter, expectedFilter)
+}
+
+// TestAssemblePeerMessage2 Tests assemblePeerMessage with messageConstNextMinDelay | messageConstUpdateRequestParams msgOps
+func TestAssemblePeerMessage2(t *testing.T) {
```

Comment:
would prefer to have longer test names that describe the test rather than TestAssemblePeerMessage2 or TestAssemblePeerMessage3